### PR TITLE
hgtv SPA

### PR DESCRIPTION
**Pull request by vidya-ram**
Changes in this PR:
1. Use Vue for client side render
2. Webpack for js & css bundling
3. vue-router for client side routing. 
4. Service worker for offline support.

### CODE REVIEW

File: hgtv/views/index.py [hunk 1]
```diff
@@ -18,9 +19,27 @@ def longdate(date):
     return utc.localize(date).astimezone(app.config['tz']).strftime('%e %B %Y')
 
 
+def jsonify_featured_channel(data):
+    channels_dict = []
```
Comment (shreyas-satish) [inline hgtv/views/index.py]: I suggest `channel_dicts`, since `channels_dict` suggests it's a `dict`.

File: hgtv/views/index.py [hunk 2]
```diff
@@ -18,9 +19,27 @@ def longdate(date):
...
+    for channel in data['channels']:
+        channels_dict.append({
+            'url': channel.url_for(),
+            'title': channel.title,
+            'logo': url_for('static', filename='thumbnails/' + channel.channel_logo_filename) if channel.channel_logo_filename else url_for('static', filename='img/sample-logo.png'),
```
Comment (shreyas-satish) [inline hgtv/views/index.py]: Please break this into multiple lines.

File: hgtv/templates/index.html.jinja2 [hunk 3]
```diff
@@ -1,101 +1,18 @@
...
-  $(document).ready(function(){
-    $('#tab-container').easytabs();
-  });
-</script>
-{% endblock %}
+{%- block appcontent %}
+	<div id="app"></div>
 
+	
+	<script src="/static/js/dist/js/manifest.951e00270cd951bf20e5.js"></script>
```
Comment (shreyas-satish) [inline hgtv/templates/index.html.jinja2]: These numbers shouldn't be hard-coded, no? They'll have to be updated every time there's a change.

File: hgtv/templates/index.html.jinja2 [hunk 4]
```diff
@@ -1,101 +1,18 @@
...
-        </div>
-      </li>
-    {% else %}
-      <p>No Channels available</p>
-    {%- endfor -%}
-  </ul>
+{% extends "home.html.jinja2" %}
+{%- block layoutheaders %}
+	
+	  <link href="/static/js/dist/css/app.a92fcd107bae02bc79c0b00cce49b4b0.css" rel="stylesheet">
```
Comment (shreyas-satish) [inline hgtv/templates/index.html.jinja2]: Indentation seems off. 2 spaces.

File: hgtv/static/js/webpack-template.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{% extends "home.html.jinja2" %}
+{%- block layoutheaders %}
+	<% for (var css in htmlWebpackPlugin.files.css) { %>
+	  <link href="<%= htmlWebpackPlugin.files.css[css] %>" rel="stylesheet">
+	<% } %>
+{% endblock %}
+
+{%- block appcontent %}
+	<div id="app"></div>
```
Comment (shreyas-satish) [inline hgtv/static/js/webpack-template.html]: Indentation seems off here too.

File: hgtv/static/js/test/unit/specs/Hello.spec.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import Vue from 'vue';
+import HelloWorld from '@/components/HelloWorld';
+
+describe('HelloWorld.vue', () => {
+  it('should render correct contents', () => {
+    const Constructor = Vue.extend(HelloWorld);
+    const vm = new Constructor().$mount();
+    expect(vm.$el.querySelector('.hello h1').textContent)
+      .to.equal('Welcome to Your Vue.js App');
```
Comment (shreyas-satish) [inline hgtv/static/js/test/unit/specs/Hello.spec.js]: Is this a dummy spec?

File: hgtv/static/js/src/components/LiveStream.vue [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+</template>
+
+<script>
+export default {
+  name: 'LiveStream',
+  props: ['livestreams'],
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (shreyas-satish) [inline hgtv/static/js/src/components/LiveStream.vue]: Remove comment?

File: hgtv/static/js/src/components/Home.vue [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  created() {
+    axios.get(this.path)
+    .then((response) => {
+      this.channels = response.data.channels;
+      this.livestreamOn = response.data.livestream.enable;
+      this.livestreams = response.data.livestream.streams;
+    })
+    .catch((e) => {
+      this.errors.push(e);
```
Comment (shreyas-satish) [inline hgtv/static/js/src/components/Home.vue]: `errors` needs to be initialized to `[]` in `data()`?

File: hgtv/views/index.py (hunk 1 above)
Comment (vidya-ram) [reply hgtv/views/index.py]: Fixed

File: hgtv/views/index.py (hunk 2 above)
Comment (vidya-ram) [reply hgtv/views/index.py]: Fixed

File: hgtv/templates/index.html.jinja2 (hunk 3 above)
Comment (vidya-ram) [reply hgtv/templates/index.html.jinja2]: This template is generated by webpack everytime we build js.

File: hgtv/static/js/src/components/LiveStream.vue (hunk 7 above)
Comment (vidya-ram) [reply hgtv/static/js/src/components/LiveStream.vue]: Fixed

File: hgtv/static/js/src/components/Home.vue (hunk 8 above)
Comment (vidya-ram) [reply hgtv/static/js/src/components/Home.vue]: Fixed

File: hgtv/static/js/test/unit/specs/Hello.spec.js (hunk 6 above)
Comment (vidya-ram) [reply hgtv/static/js/test/unit/specs/Hello.spec.js]: Yes

**Comment by vidya-ram** [comment]
Opening a new PR

**Pull request closed without merging**